include/miral/miral/wayland_extensions.h

```diff
@@ -163,16 +188,24 @@ class WaylandExtensions
     /// \remark Since MirAL 2.6
     static auto supported() -> std::set<std::string>;
 
-    /// Enable a Wayland extension
-    /// Throws a std::runtime_error if the extension is not supported
+    /// Enable a Wayland extension by default. The user can still disable it with the drop-wayland-extensions or
+    /// wayland-extensions options. The extension can be forced to be enabled regardless of user options with
+    /// conditionally_enable().
     /// \remark Since MirAL 2.6
     auto enable(std::string name) -> WaylandExtensions&;
 
-    /// Disable a Wayand extension
-    /// Throws a std::runtime_error if the extension is not supported
+    /// Disable a Wayland extension by default. The user can still enable it with the add-wayland-extensions or
+    /// wayland-extensions options. The extension can be forced to be disabled regardless of user options with
+    /// conditionally_enable().
     /// \remark Since MirAL 2.6
     auto disable(std::string name) -> WaylandExtensions&;
 
+    /// Enable a Wayland extension only when the callback returns true. The callback can use info.user_preference()
+    /// to respect the extension options the user provided, it is not required. Unlike enable() and disable(),
+    /// conditionally_enable() can override the user options.
```

Comment:
[Not blocking, but a note]

What we discussed on Mattermost was:

> Following your suggestion above, for conditionally_enable(), we should probably not callback if the config is used to enable a protocol.

This proposal may well be a better approach, but we should choose explicitly